### ISSUE TITLE
Using `fit_resamples()` with `validation_set()` produces NA warning

### ISSUE THREAD

**Issue by EmilHvitfeldt** [tidymodels/tune#769]
``` r
library(tidymodels)

set.seed(6735)
folds <- initial_validation_split(mtcars) %>%
  validation_set()

spline_rec <- recipe(mpg ~ ., data = mtcars) %>%
  step_ns(disp) %>%
  step_ns(wt)

lin_mod <- linear_reg() %>%
  set_engine("lm")

fit_resamples(lin_mod, spline_rec, folds)
#> Warning in `[.tbl_df`(x, is.finite(x <- as.numeric(x))): NAs introduced by
#> coercion
#> # Resampling results
#> #  
#> # A tibble: 1 × 4
#>   splits         id         .metrics         .notes          
#>   <list>         <chr>      <list>           <list>          
#> 1 <split [19/6]> validation <tibble [2 × 4]> <tibble [0 × 3]>
```

**Comment by simonpcouch** [comment]
Related to #731 and https://github.com/tidymodels/rsample/issues/456#issue-1919309763.

**Comment by simonpcouch** [comment]
Closed in tidymodels/rsample#456. :)
